apps/cowswap-frontend/src/modules/tradeWidgetAddons/containers/MetamaskTransactionWarning/index.tsx

```diff
@@ -81,12 +69,92 @@ function useWidgetProviderMetaInfo() {
 }
 
 /**
- * This is hacky way to detect if the wallet is metamask mobile injected wallet
- * Many injected wallet browsers emulate isMetaMask, but only metamask mobile has _metamask
+ * Fetch the Metamask version using the method defined in https://docs.metamask.io/wallet/reference/json-rpc-methods/web3_clientversion
+ * Returns null if the version could not be fetched
  */
-function useIsMetamaskMobileInjectedWallet(): boolean {
-  const walletProvider = useWalletProvider()
-  const rawProvider = walletProvider?.provider as any
+async function getMetamaskVersion(provider: any): Promise<string | null> {
```

Comment:
`any` type is OK here or we need it more explicitly type? (ExternalProvider?)

Reply:
You are right, I was just being lazy.
Will updated.